src/OSPSuite.Core/Domain/ObjectPath.cs

```diff
@@ -146,6 +149,15 @@ public virtual T Resolve<T>(IEntity refEntity) where T : class
          return resolvePath<T>(dependentObject, usePath);
       }
 
+      private void removeMatchingPaths(IContainer root, List<string> usePath)
+      {
+         var matchingElements = !string.IsNullOrEmpty(root?.ParentPath?.PathAsString) &&
+                                new ObjectPath(_pathEntries).PathAsString.StartsWith(root?.ParentPath?.PathAsString ?? string.Empty);
```

Comment:
the part after `&&` could be simplified to 
`new ObjectPath(_pathEntries).PathAsString.StartsWith(root.ParentPath.PathAsString);`
(because the part before `&&` already ensures that `root.ParentPath.PathAsString` is not null and not empty)